extensions/resource-deployment/src/ui/notebookWizard/notebookWizardModel.ts

```diff
@@ -58,16 +59,27 @@ export class NotebookWizardModel extends ResourceTypeModel {
 	}
 
 	/**
-	 * Generates the notebook and returns true on successful generation
+	 * Generates the notebook and returns true if generation was done and so the wizard should be closed.
 	 **/
 	public async onGenerateScript(): Promise<boolean> {
 		const lastPage = this.wizard.lastPage! as NotebookWizardPage;
 		if (lastPage.validatePage()) {
-			const notebook = await this.prepareNotebookAndEnvironment();
-			await this.openNotebook(notebook);
-			return true;
+			let notebook: Notebook;
+			try {
+				notebook = await this.prepareNotebookAndEnvironment();
+			} catch (e) {
+				const isUserCancelled = (e: any) => e instanceof Error && 'type' in e && (<ErrorWithType>e).type === ErrorType.userCancelled;
```

Comment:
Just use this logic directly - if you're not re-using the function no reason to add the extra overhead to create the function object here.

Reply:
makes the code more readable. The negated condition is harder to read otherwise

Reply:
I could make isUserCancelled a variable instead of function perhaps.

Reply:
Done. Changed isUserCancelled to a constant